### PR TITLE
Old god wonders

### PR DESCRIPTION
**Pull request by sejtalk**
## Changelog:
- Added Ahn'Qiraj wonder with Silithid Greathive upgrade.
- Added Shrine of Storms wonder with Crucible of Storms upgrade.
- Added Ulduar wonder with Forge of  Wills upgrade.
- Added Titan-forged retinue.

### CODE REVIEW

**Comment by sejtalk** [comment]
> Ahn'Qiraj and the Shrine of Storms are good. However, I couldn't see the Ulduar wonder (let alone the upgrade). The province was a wasteland for me.

Thats's because it only adds Ulduar wonder not the province itself.

**Comment by LordSilvermane** [comment]
Ahh. Any way for me to see the wonder without the province?

**Comment by sejtalk** [comment]
> Ahh. Any way for me to see the wonder without the province?

You would need to edit history file of some province and add it there along with the upgrade. And i guess you could just check the code.

File: common/wonders/wc_wonders_historical.txt [hunk 1]
```diff
@@ -247,3 +247,396 @@ wonder_wyrmrest_temple = {
...
+		local_modifier = {
+		}
+		owner_modifier = {
+			monthly_character_prestige = 0.2
+			monthly_character_piety = 0.2
+		}
+	}
+}
+# Ulduar
+wonder_ulduar = {
```
Comment (zumbak04) [inline common/wonders/wc_wonders_historical.txt]: It doesn't exist in the game, so there's no point in adding it now.

**Comment by zumbak04** [comment]
> You would need to edit history file of some province and add it there along with the upgrade. And i guess you could just check the code.

@Warcraft-GoA-Development-Team/keepers except me and @MemeWizard can't script.

File: common/wonders/wc_wonders_historical.txt (hunk 1 above)
Comment (sejtalk) [reply common/wonders/wc_wonders_historical.txt]: South seas branch merge when?

Comment (zumbak04) [reply common/wonders/wc_wonders_historical.txt]: No idea.

Comment (MemeWizard) [reply common/wonders/wc_wonders_historical.txt]: You can move this wonder to separate branch in order to merge Silithid Greathive and Shrine of Storms.
And when south seas branch will be merged, create PR for Ulduar.

File: common/retinue_subunits/00_retinue_subunits.txt [hunk 2]
```diff
@@ -1112,7 +1118,13 @@ wc_ret_cul_naga = {
 
 	potential = {
 		# is_tribal = no
-		culture = naga
+		OR = {
+			culture = qiraji
```
Comment (zumbak04) [inline common/retinue_subunits/00_retinue_subunits.txt]: You must have qiraji culture to unlock nagas.

File: interface/wc_wonder.gfx [hunk 3]
```diff
@@ -98,4 +98,70 @@ spriteTypes = {
...
+		name = "GFX_wonder_strip_shrine_storm_small"
+		texturefile = "gfx\\interface\\wonder_interface\\wonder_strips\\wonder_strip_shrine_of_storms_small.dds"
+		noOfFrames = 5
+	}
+	# wonder_shrine_storm
+	spriteType = {
+		name = "GFX_wonder_strip_ulduar_large"
+		texturefile = "gfx\\interface\\wonder_interface\\wonder_strips\\wonder_strip_ulduar_large.dds"
+		noOfFrames = 5
+	}
```
Comment (zumbak04) [inline interface/wc_wonder.gfx]: The comment lies. It's Ulduar.

File: localisation/00_wc_sad.csv [hunk 4]
```diff
@@ -773,3 +773,6 @@ whispers_option_stewardship;
 whispers_option_intrigue;�Y[Root.GetFirstName]�! will focus on writing down whispers about �YIntrigue�!.;;;;;;;;;;;;;x
 whispers_option_learning;�Y[Root.GetFirstName]�! will focus on writing down whispers about �YLearning�!.;;;;;;;;;;;;;x
 nick_the_corruptor;the Corruptor;;[From.Getderdie] Verderber[From.GetinFemale];;;;;;;;;;;x
+text_effect_qiraji_retinue;Enables �YSilithid Swarm�! retinue avaible to �YQiraji�! culture.;;Aktiviert das �YSilithiden-Schwarm�!-Gefolge f�r die �YQiraji�!-Kultur.;;;;;;;;;;;x
+text_effect_naga_retinue;Enables �YRoyal Guard�! retinue avaible to �YNaga�! culture.;;Aktiviert das �YK�nigswache�!-Gefolge f�r die �YNaga�!-Kultur.;;;;;;;;;;;x
+text_effect_titanforged_retinue;Enables �YTitan-forged Legion�! retinue avaible to �YTitan-forged�! culture group.;;Aktiviert das �YTitanische Legion!-Gefolge f�r die �Ytitanische�! Kultur.;;;;;;;;;;;x
```
Comment (zumbak04) [inline localisation/00_wc_sad.csv]: Typo in available. Also, it sounds like you must have these cultures to unlock them.

File: common/wonder_upgrades/wc_wonders_historical_upgrades.txt [hunk 5]
```diff
@@ -1 +1,69 @@
...
+	gold_cost_ticking = 1
+	local_modifier = {
+	}
+	owner_modifier = {
+		land_organisation = 0.05
+		retinuesize = 250
+		retinuesize_perc = 0.2
+		text_effect_void_inflow = 1
+		text_effect_naga_retinue = 1
+	}
```
Comment (zumbak04) [inline common/wonder_upgrades/wc_wonders_historical_upgrades.txt]: The best upgrade. Gives everything that other upgrades give + Void Inflow + `land_organisation`.

@DukeJeans had [Documentation](https://github.com/Warcraft-GoA-Development-Team/Warcraft-Guardians-of-Azeroth/wiki/Links-to-External-Documentation) to balance wonders, check it out. Not sure how it works though. Maybe it's better to use the balancing codex at this point.

Comment (sejtalk) [reply common/wonder_upgrades/wc_wonders_historical_upgrades.txt]: Check the other wonder upgrades again. They have local modifier buffs

Comment (zumbak04) [reply common/wonder_upgrades/wc_wonders_historical_upgrades.txt]: The problem that they're local while `land_organisation` affects more than one province/army/character, etc. So, we need to come up with some balance.

Comment (zumbak04) [reply common/wonder_upgrades/wc_wonders_historical_upgrades.txt]: I feel it like `land_organisation` is too strong and `text_effect_void_inflow` more or less (most likely more) costs like local modifiers of other wonders.

Comment (sejtalk) [reply common/wonder_upgrades/wc_wonders_historical_upgrades.txt]: Getting +100% garrison on your old god province is much better than void inflow imo

Comment (sejtalk) [reply common/wonder_upgrades/wc_wonders_historical_upgrades.txt]: And land organization and void inflow are from one of wonder_upgrades

Comment (zumbak04) [reply common/wonder_upgrades/wc_wonders_historical_upgrades.txt]: > Getting +100% garrison on your old god province is much better than void inflow imo

Maybe you're right, void inflow gives a lot, but it's random. But `garrison_size` is on the same level, to be honest. It just makes you sure the enemy needs more troops to assault your capital.

Comment (sejtalk) [reply common/wonder_upgrades/wc_wonders_historical_upgrades.txt]: well the doc by @DukeJeans shows that my modifiers are pretty balanced.

Comment (zumbak04) [reply common/wonder_upgrades/wc_wonders_historical_upgrades.txt]: Hmm? I don't see them in the doc.

Comment (sejtalk) [reply common/wonder_upgrades/wc_wonders_historical_upgrades.txt]: I used the doc to look at what modifiers are used there

File: localisation/00_wc_sad.csv [hunk 6]
```diff
@@ -705,3 +705,6 @@ whispers_option_stewardship;
 whispers_option_intrigue;�Y[Root.GetFirstName]�! will focus on writing down whispers about �YIntrigue�!.;;;;;;;;;;;;;x
 whispers_option_learning;�Y[Root.GetFirstName]�! will focus on writing down whispers about �YLearning�!.;;;;;;;;;;;;;x
 nick_the_corruptor;the Corruptor;;[From.Getderdie] Verderber[From.GetinFemale];;;;;;;;;;;x
+text_effect_qiraji_retinue;Enables �YSilithid Swarm�! retinue that is normally available to �YQiraji�! culture only.;;Aktiviert das �YSilithiden-Schwarm�!-Gefolge f�r die �YQiraji�!-Kultur.;;;;;;;;;;;x
+text_effect_naga_retinue;Enables �YRoyal Guard�! retinue that is normally available to �YNaga�! culture only.;;Aktiviert das �YK�nigswache�!-Gefolge f�r die �YNaga�!-Kultur.;;;;;;;;;;;x
+text_effect_titanforged_retinue;Enables �YTitan-forged Legion�! retinue that is normally available to �YTitan-forged�! culture group only.;;Aktiviert das �YTitanische Legion!-Gefolge f�r die �Ytitanische�! Kultur.;;;;;;;;;;;x
```
Comment (zumbak04) [inline localisation/00_wc_sad.csv]: I would remove dots and paint the retinue names green. Usually, effects don't have dots at the end.
<details>
<summary>Click to expand</summary>

image

</details>

File: common/wonders/wc_wonders_historical.txt [hunk 7]
```diff
@@ -247,3 +247,403 @@ wonder_wyrmrest_temple = {
...
+				religion = old_gods_worship
+				religion = mystery_of_the_makers
+				religion = cult_of_loken
+			}
+		}
+	}
+	flags = {
+		defensive
+		cost_medium
+	}
```
Comment (zumbak04) [inline common/wonders/wc_wonders_historical.txt]: Don't have historical flag.

File: common/wonders/wc_wonders_historical.txt [hunk 8]
```diff
@@ -247,3 +247,403 @@ wonder_wyrmrest_temple = {
...
+	available_upgrades = {
+		# VERY UNIQUE:
+		upgrade_forge_of_wills
+		upgrade_twilight_altar # Taken from wonder_twilight_citadel
```
Comment (zumbak04) [inline common/wonders/wc_wonders_historical.txt]: The Shrine of Storms don't have it, is it ok?

File: common/wonders/wc_wonders_historical.txt [hunk 9]
```diff
@@ -247,3 +247,403 @@ wonder_wyrmrest_temple = {
...
+wonder_ulduar = {
+	allow_construction = no
+	picture = GFX_wonder_strip_ulduar
+	potential = {
+		OR = {
+			religion = old_gods_worship
+			religion = mystery_of_the_makers
+			religion = cult_of_loken
+		}
+	}
```
Comment (zumbak04) [inline common/wonders/wc_wonders_historical.txt]: Why it's not empty if you can't build it?

File: common/wonders/wc_wonders_historical.txt [hunk 10]
```diff
@@ -247,3 +247,403 @@ wonder_wyrmrest_temple = {
...
+wonder_shrine_storm = {
+	allow_construction = no
+	picture = GFX_wonder_strip_shrine_storm
+	model = wonder_temple
+	potential = {
+		OR = {
+			religion = old_gods_worship
+			religion = tidemother
+		}
+	}
```
Comment (zumbak04) [inline common/wonders/wc_wonders_historical.txt]: Why it's not empty if you can't build it?

File: common/wonders/wc_wonders_historical.txt [hunk 11]
```diff
@@ -247,3 +247,403 @@ wonder_wyrmrest_temple = {
...
+		}
+		owner_modifier = {
+			monthly_character_prestige = 0.2
+			monthly_character_piety = 0.2
+		}
+	}
+}
+# Ulduar
+# Based on wonder_fortress
+wonder_ulduar = {
```
Comment (zumbak04) [inline common/wonders/wc_wonders_historical.txt]: I don't know why, but Ulduar gives bonuses even when not active.
<details>
<summary>Click to expand</summary>

image

</details>

The Sunwell from the same file doesn't do it.
<details>
<summary>Click to expand</summary>

image

</details>

Comment (sejtalk) [reply common/wonders/wc_wonders_historical.txt]: no idea why. will try to find out whats wrong

**Pull request merged**